client/settings/fraud-protection/advanced-settings/rule-card-notice.tsx

```diff
@@ -11,7 +11,14 @@ import './../style.scss';
 import BannerNotice from 'wcpay/components/banner-notice';
 import { TipIcon } from 'wcpay/icons';
 
-const FraudProtectionRuleCardNotice = ( { type, children } ) => {
+interface FraudProtectionRuleCardNoticeProps {
+	type: any;
+}
```

Comment:
We ended up going with 'any', b/c if we use `'error' | 'warning' | 'info'` we end up with an error on this test that we had to modify to accpt an 'invalid' type. Before it was testing if no type was passed and TS didn't like that. 
https://github.com/Automattic/woocommerce-payments/blob/ed593379de71cdb8e7c3c76645681830b64d4839/client/settings/fraud-protection/advanced-settings/test/rule-card-notice.test.tsx#L38

I'll leave it as 'any' for now but if you have any other suggestions, I'm all ears.

Reply:
There are different ways to do it, an example can be:

```tsx
// rule-card-notice.tsx
const supportedTypes = [ 'error', 'warning', 'info' ] as const; // `const` will make its type literal array.

interface FraudProtectionRuleCardNoticeProps {
	type: typeof supportedTypes[ number ]; // With `typeof` and `[number]` we get the template literal.
}

// rule-card-notice.test.tsx
<FraudProtectionRuleCardNotice type={ 'invalid' as any
/* You can cast to any to avoid any TS error and test it,
emulating the behavior of using the component from a JS file. */  }>
```

Reply:
@eduardoumpierre helped me with this a few days ago and he came up with the same solution. Great minds think alike. 😁